### PR TITLE
Functional test for tensorflow multi stream with VGPUs

### PR DESCRIPTION
**Pull request by xprotobeast2**

### CODE REVIEW

File: qa/L0_tf_multi_stream/multi_stream_client.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/python
+
+# Copyright (c) 2018, NVIDIA CORPORATION. All rights reserved.
```
Comment (deadeyegoodwin) [inline qa/L0_tf_multi_stream/multi_stream_client.py]: 2019

File: qa/L0_tf_multi_stream/multi_stream_client.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+    # Create the data for the input tensor.
+    input_data = np.array([FLAGS.delay], dtype=np.int32)
+
+    # Send N inference requests to the inference server. Time the inference for both 
+    # requests
+    start_time = time()
+
+    for i in range(FLAGS.count):
+        infer_ctx.async_run_with_cb(partial(completion_callback), { 'in' : (input_data,) }, { 'out' : InferContext.ResultFormat.RAW }, batch_size)
```
Comment (deadeyegoodwin) [inline qa/L0_tf_multi_stream/multi_stream_client.py]: too long, wrap this line

File: qa/L0_tf_multi_stream/multi_stream_client.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    FLAGS = parser.parse_args()
+    protocol = ProtocolType.from_str(FLAGS.protocol)
+
+    # We use model that takes 1 input tensor containing the delay number of cycles
+    # to occupy an SM
+    model_name = FLAGS.model
+    model_version = 1
+    batch_size = 1
+
+    # Create a health context, get the ready and live state of server.
```
Comment (deadeyegoodwin) [inline qa/L0_tf_multi_stream/multi_stream_client.py]: I don't think you need to get health and status for this test, do you?

File: qa/L0_tf_multi_stream/multi_stream_client.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    while True:
+        request_id, finish_time = finish_times.get()
+        result = infer_ctx.get_async_run_results(request_id, True)
+        finished_requests += 1
+        print("Request %d"%request_id + " finished in %f"%finish_time)
+        max_completion_time = max(max_completion_time, finish_time)
+        if (finished_requests == FLAGS.count):
+            break
+    
+    print("Completion time for %d instances: %f secs"%(i+1, max_completion_time))
```
Comment (deadeyegoodwin) [inline qa/L0_tf_multi_stream/multi_stream_client.py]: need newline at end of file

File: qa/L0_tf_multi_stream/test.sh [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    run_server
+    if [ "$SERVER_PID" == "0" ]; then
+        echo -e "\n***\n*** Failed to start $SERVER\n***"
+        cat $SERVER_LOG
+        exit 1
+    fi
+
+    # The first run of the client warms up TF/CUDA
+    set +e
+    python $MULTI_STREAM_CLIENT -v -i grpc -u localhost:8001 -m $MODEL -c $INSTANCE_CNT -d 2100000000 >> /dev/null
```
Comment (deadeyegoodwin) [inline qa/L0_tf_multi_stream/test.sh]: Can you put 210000... into a documented variable

File: qa/L0_tf_multi_stream/test.sh [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    python $MULTI_STREAM_CLIENT -v -i grpc -u localhost:8001 -m $MODEL -c $INSTANCE_CNT -d 2100000000 >> $CLIENT_LOG 2>&1
```
Comment (deadeyegoodwin) [inline qa/L0_tf_multi_stream/test.sh]: -d does a sweep over different concurrencies... you should just do "-t $INSTANCE_CNT" to run perf-client at a fixed concurrency. Don't need -c.

File: src/backends/tensorflow/tensorflow_backend_tf.cc [hunk 7]
```diff
@@ -175,6 +175,17 @@ NewSessionOptions(
       ->set_per_process_gpu_memory_fraction(per_process_gpu_memory_fraction);
   session_options->config.set_allow_soft_placement(allow_soft_placement);
 
+
+
+  if(std::getenv("TRTIS_TF_INTER_OP_THREADS") && std::getenv("TRTIS_TF_INTRA_OP_THREADS")) {
```
Comment (deadeyegoodwin) [inline src/backends/tensorflow/tensorflow_backend_tf.cc]: I don't think this should be in this commit.

File: qa/L0_tf_multi_stream/test.sh (hunk 6 above)
Comment (xprotobeast2) [reply qa/L0_tf_multi_stream/test.sh]: Do you mean that I should use different arguments to the script I wrote? This isn't perf_client

File: qa/L0_tf_multi_stream/test.sh [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+    run_server
+    if [ "$SERVER_PID" == "0" ]; then
+        echo -e "\n***\n*** Failed to start $SERVER\n***"
+        cat $SERVER_LOG
+        exit 1
+    fi
+
+    # The first run of the client warms up TF/CUDA
+    set +e
+    python $MULTI_STREAM_CLIENT -v -i grpc -u localhost:8001 -m $MODEL -c $INSTANCE_CNT -n $NUM_DELAY_CYCLES >> /dev/null
```
Comment (deadeyegoodwin) [inline qa/L0_tf_multi_stream/test.sh]: Don't use -n.
-c only has meaning with -d, you want to use -t. You should verify that perf_client is doing the right thing

File: qa/common/busy_op_kernel.cc [hunk 9]
```diff
@@ -0,0 +1,43 @@
+#include <time.h>
```
Comment (deadeyegoodwin) [inline qa/common/busy_op_kernel.cc]: Need copyright on this file.

File: qa/common/busy_op_kernel.cu.cc [hunk 10]
```diff
@@ -0,0 +1,45 @@
+#if GOOGLE_CUDA
```
Comment (deadeyegoodwin) [inline qa/common/busy_op_kernel.cu.cc]: Need copyright on this file.

**Pull request merged**